crates/ruff_linter/src/rules/refurb/rules/slice_to_remove_prefix_or_suffix.rs

```diff
@@ -0,0 +1,498 @@
+                        compr_affix == compr_arg
+                    })
+            },
+        ),
+        _ => false,
+    }
+}
+
+/// Generates the source code string
+/// ```python
+/// text = text.removeprefix(prefix)
+/// ```
+/// or
+/// ```python
+/// text = text.removesuffix(prefix)
+/// ```
+/// as appropriate.
+fn generate_assignment_with_removeaffix(
+    text: &str,
+    affix_query: &AffixQuery,
+    generator: Generator,
+) -> String {
+    let remove_affix_expr = make_removeaffix_expr(text, affix_query);
+    generator.stmt(&ast::Stmt::Assign(ast::StmtAssign {
+        range: TextRange::default(),
+        targets: vec![ast::Expr::Name(ast::ExprName {
+            range: TextRange::default(),
+            id: ast::name::Name::from(text),
+            ctx: ast::ExprContext::Store,
+        })],
+        value: Box::new(remove_affix_expr),
+    }))
+}
+
+/// Generates the source code string
+/// ```python
+/// text.removeprefix(prefix)
+/// ```
+/// or
+///
+/// ```python
+/// text.removesuffix(suffix)
+/// ```
+/// as appropriate.
+fn generate_removeaffix_expr(text: &str, affix_query: &AffixQuery, generator: Generator) -> String {
+    let remove_affix_expr = make_removeaffix_expr(text, affix_query);
+    generator.expr(&remove_affix_expr)
+}
+
+/// Creates the AST node corresponding to
+/// ```python
+/// text.removeprefix(prefix)
+/// ```
+/// or
+///
+/// ```python
+/// text.removesuffix(suffix)
+/// ```
+/// as appropriate.
+fn make_removeaffix_expr(text: &str, affix_query: &AffixQuery) -> ast::Expr {
+    let func_name = match affix_query.kind {
+        AffixKind::StartsWith => ast::name::Name::from("removeprefix"),
+        AffixKind::EndsWith => ast::name::Name::from("removesuffix"),
+    };
+    ast::Expr::Call(ast::ExprCall {
+        range: TextRange::default(),
+        func: Box::new(ast::Expr::Attribute(ast::ExprAttribute {
+            range: TextRange::default(),
+            value: Box::new(ast::Expr::Name(ast::ExprName {
+                range: TextRange::default(),
+                id: ast::name::Name::from(text),
+                ctx: ast::ExprContext::Store,
+            })),
+            attr: ast::Identifier {
+                id: func_name,
+                range: TextRange::default(),
+            },
+            ctx: ast::ExprContext::Load,
+        })),
+        arguments: ast::Arguments {
+            range: TextRange::default(),
+            args: Box::new([affix_query.affix.clone()]),
+            keywords: Box::new([]),
+        },
+    })
+}
+
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+enum StmtOrExpr {
+    Statement,
+    Expression,
+}
+
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+enum AffixKind {
+    StartsWith,
+    EndsWith,
+}
+#[derive(Debug)]
+struct AffixQuery<'a> {
+    kind: AffixKind,
+    affix: &'a ast::Expr,
+}
+
+#[derive(Debug)]
+struct RemoveAffixData<'a> {
+    text: &'a str,
+    bound: &'a ast::Expr,
+    affix_query: AffixQuery<'a>,
+}
```

Comment:
nit: some docstrings for the fields on these structs would be nice ;)

Reply:
Minimal docstrings added